web/app.js

```diff
@@ -2001,6 +2001,12 @@
         opts
       );
     }
+    window.addEventListener("message", event => {
```

Comment:
We need to add origin check for better security, but not sure how to add it while still make it testable in development, review app and prod at the same time.

Reply:
A first step is to validate the origin and allow *.herokuapp.com and *.sharinpix.com.

Reply:
Its already not totally safe to allow all `.herokuapp.com`, but i allowed `https://localhost:4200/pagelayout` also for now, so it will work in ember app also in local. Its better than allow all, need to find a better security later.